cmd/age/age.go

```diff
@@ -455,10 +455,20 @@ func decryptNotPass(flags identityFlags, in io.Reader, out io.Writer) {
 }
 
 func decryptPass(in io.Reader, out io.Writer) {
+	passphrase := passphrasePromptForDecryption
+
+	if in != os.Stdin && !term.IsTerminal(int(os.Stdin.Fd())) {
+		passphrase = func() (string, error) {
+			b, err := io.ReadAll(os.Stdin)
+			b = bytes.TrimRight(b, "\n")
```

Comment:
Trimming newline to make test case work and improve usability when passphrase is coming from a file or via echo.